src/s3/activation.ts

```diff
@@ -29,7 +31,16 @@ export async function activate(extensionContext: vscode.ExtensionContext): Promi
             await downloadFileAsCommand(node)
         }),
         vscode.commands.registerCommand('aws.s3.uploadFile', async (node: S3BucketNode | S3FolderNode) => {
-            await uploadFileCommand(node)
+            const regionCode = DEFAULT_REGION
```

Comment:
This is a slight regression in that S3 nodes will use the DEFAULT_REGION for uploads instead of the node's region. That being said, the S3 nodes are created by passing a client rather than what we did with previous node types (passing a region), which means that you have to find the parent node for the region code.

I'm thinking sending the region code to the node level is out of scope for this PR, but could you add a TODO to use the S3 node's region code here?

Reply:
What if I just change the call when a node was given to `await uploadFileCommand(node.s3, node)` and put the logic of creating the s3client instance inside the `if (!node)`? However, doing that I would need to keep the s3 client inside S3FolderNode and S3BucketNode as public

Reply:
I like what you proposed here for now. In the future, we may want to go towards the other version so we can make the field private and also have region context at the node level, but for now this works and this way we don't have to create any additional clients. Thoughts @JadenSimon ?

Reply:
Yeah that works. The client here isn't being mutated by the bucket anyway so making it public is not a big deal.